### ISSUE TITLE
Close price different between methods & profit simulator

### ISSUE THREAD

**Issue by igtk** [askmike/gekko#427]
Hello,

According to what I think the price should be equally.
When I compared close price in an method with close price from Profit simulator, the result is
if candleSize = 1 Profit simulator **always** based on the **last** candle to calculate profit 
if candleSize = 2 Profit simulator **often** based on the **current** candle to calculate profit
if candleSize = x (Profit simulator candle.close) **!=** (method candle.close)

Example when candleSize = 1; 
2016-08-21 10:15:45 (INFO):      Last Open price: 0.01880198       //method
2016-08-21 10:15:45 (INFO):      Last Close price: 0.01877336  
2016-08-21 10:15:45 (INFO):      Last High price: 0.01880542
2016-08-21 10:15:45 (INFO):      Last Low price: 0.01876003
2016-08-21 10:15:45 (INFO):      Open price: 0.01877849
2016-08-21 10:15:45 (INFO):      Close price: 0.01876001
2016-08-21 10:15:45 (INFO):      High price: 0.01877988
2016-08-21 10:15:45 (INFO):      Low price: 0.01876001
2016-08-21 10:15:45 (INFO): Profit simulator got advice to short    @ 2016-08-18 19:34:01, selling 0 ETH    (98.03533514)
2016-08-21 10:15:45 (INFO):      Profit simulator Open price: 0.01880198       //Profit simulator
2016-08-21 10:15:45 (INFO):      Profit simulator Close price: 0.01877336
2016-08-21 10:15:45 (INFO):      Profit simulator High price: 0.01880542
2016-08-21 10:15:45 (INFO):      Profit simulator Low price: 0.01876003

**_Notes:_** 
All the tests that I've made was in backtest mode.
Of course I've modified the method and Profit simulator files to print these additional lines using log.info() function.

**Comment by askmike** [comment]
Great that you are diving into some things in Gekko! What do you think is the desired behaviour here? The real trader just listens to the advice and only then figures out what the current market price is. The closest to that would be for the profit simulator to use the best available price, or not?

**Comment by igtk** [comment]
Hello, mike.
I made another tests, and I found this result;
https://paste.debian.net/plain/790879

Profit simulator this.price (candle.close) = the last candle.close. (it's ok)
candle.close variable from method file = First **next** 1 min candle.close. but I think it should be the same as the Last 1 min candle.close not the next ,like Profit Simulator.

eg: when i write method depend on the current market price to decide long or short, 
The closest available price I think is the **Last** 1 min candle.close, but candle.close in method currently refer to the **Next** 1 min candle.close!!! (in real it's impossible).
